### DISCUSSION TITLE
Improve the look of my music sheet

### DISCUSSION THREAD

**Discussion by VadeInPace** [ImageMagick/ImageMagick#6402] [Help] [answered]
### ImageMagick version

7.1.1-4 Q16-HDRI

### Operating system, version and so on

Linux

### Description

Hello there,

Is it possible to improve the drawing of this music sheet sample.
![Musicsample](https://github.com/ImageMagick/ImageMagick/assets/49396694/ae65ac1c-d0d0-4fdd-94bb-571f05499278)
When I print it stave lines are hard to see.

Thanks

**Comment by fmw42** [comment, marked as answer]
`magick music.png -morphology open rectangle:7x5 -auto-threshold otsu x.png`

![x](https://github.com/ImageMagick/ImageMagick/assets/13458218/c1e01d02-bc28-48c1-a5f7-47bbb20a35a6)

**Reply by VadeInPace** [reply]
Thank you so much, a good improvement.
For my knowledge, could you provide a link or quickly explain what do 
* rectangle:AxB (behaviour of A and B)
* -auto-threshold otsu.

**Comment by fmw42** [comment]
AxB is the size of the rectangular morphology filters.  See https://imagemagick.org/Usage/morphology/#rectangle

-auto-threshold otsu is the auto-threshold method.  See https://imagemagick.org/script/command-line-options.php#auto-threshold. See also https://en.wikipedia.org/wiki/Otsu%27s_method and https://www.baeldung.com/cs/otsu-segmentation and https://homepages.inf.ed.ac.uk/rbf/CVonline/LOCAL_COPIES/MORSE/threshold.pdf
